compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/BallerinaUserHome.java

```diff
@@ -36,6 +37,8 @@ private BallerinaUserHome(Environment environment, Path ballerinaUserHomeDirPath
                 .resolve(ProjectConstants.CENTRAL_REPOSITORY_CACHE_NAME);
         try {
             Files.createDirectories(remotePackageRepositoryPath);
+        } catch (AccessDeniedException ae) {
+            throw new ProjectException("permission denied: " + ae.getMessage());
```

Comment:
```
Failed to create /home/user/.ballerina/Settings.toml: permission denied
```
Shall we give the error with the path? How this error will be printed now? Can you add that to the PR as well?